src/data_structure/board.java

```diff
@@ -0,0 +1,49 @@
+package data_structure;
+
+import java.util.Vector;
+
+public class board {
+    private byte width;
+    private byte height;
+
+    private field2D[][] gameMap;
+    private Vector<fieldInfo> columnsInfo;
+    private Vector<fieldInfo> rowsInfo;
+
+    public board(byte width, byte height, field2D[] startMap) {
```

Comment:
Why `startMap` is only 1D array? I think 2D array should be better and simpler to use.